src/box/lua/net_box.lua

```diff
@@ -1113,6 +1196,50 @@ local function check_eval_args(args)
     end
 end
 
+function stream_methods:stream()
+    check_remote_arg(self, 'stream')
+    return self
+end
+
+function remote_methods:stream(stream_id)
+    check_remote_arg(self, 'stream')
+    if type(stream_id) ~= 'number' and type(stream_id) ~= 'nil' then
+        error("Use remote:stream(stream_id) with an integer stream_id or " ..
+              "without any arguments")
+    end
+    local last_sid = self._last_stream_id
+    if (stream_id and last_sid > 0) or (not stream_id and last_sid < 0) then
+        error("You can use stream_id, which is generated automatically, " ..
+              "or set it manually, but do not mix these two options")
+    end
+    self._last_stream_id = stream_id and -1 or (last_sid + 1)
+    stream_id = stream_id and stream_id or self._last_stream_id
+
+    if self._streams[stream_id] then
+        return self._streams[stream_id]
+    end
+
+    local stream = setmetatable({
+        _stream_id = stream_id,
+        stream = stream_methods.stream,
+        space = setmetatable({
+            _proxy = nil,
+        }, proxy_stream_space_mt),
+        _src = self,
+    }, { __index = self })
+    stream.space._proxy = stream
+    -- On each schema reload we reload spaces for
+    -- each stream with any access to space.
+    self:on_schema_reload(function()
+        if stream._space_was_accessed then
+            stream.space = setmetatable({
+                _proxy = stream,
+            }, proxy_stream_space_mt)
```

Comment:
The problem of the trigger is that it is set for all streams, even if they do not use spaces. Another problem is that all the streams leak now. The connections too. I tried this test:
```Lua
netbox = require('net.box')
box.cfg{listen = 3313}
box.schema.user.grant('guest', 'super')

c = netbox.connect(box.cfg.listen)
for i = 1, 10 do c:stream() end
collectgarbage()
c = nil
collectgarbage()
```
And added this diff:
```Diff
@@ -1226,6 +1226,9 @@ function remote_methods:stream(stream_id)
             _proxy = nil,
         }, proxy_stream_space_mt),
         _src = self,
+        ffi.gc(ffi.new('char[1]'), function()
+            print('stream delete')
+        end)
```
The message in the test above is never printed. I also tried to create the streams in a loop with 1000000 iterations and GC every 10k iterations. Eventually I run out of memory and crashed.

Reply:
Fixed, and add test for it